runtime/compiler/runtime/RelocationRecord.cpp

```diff
@@ -5684,6 +5687,74 @@ TR_RelocationRecordMethodPointer::activatePointer(TR_RelocationRuntime *reloRunt
       registerHCRAssumption(reloRuntime, reloLocation);
    }
 
+char *
+TR_RelocationRecordInlinedMethodPointer::name()
+   {
+   return "TR_InlinedMethodPointer";
+   }
+
+void
+TR_RelocationRecordInlinedMethodPointer::preparePrivateData(TR_RelocationRuntime *reloRuntime, TR_RelocationTarget *reloTarget)
+   {
+   TR_RelocationRecordPointerPrivateData *reloPrivateData = &(privateData()->pointer);
+      J9Method *inlinedMethod =
+            reinterpret_cast<J9Method *>(
+               getInlinedSiteMethod(reloRuntime, inlinedSiteIndex(reloTarget)));
+
+   if (inlinedMethod != reinterpret_cast<J9Method *>(-1))
+      {
+      reloPrivateData->_activatePointer = true;
+      reloPrivateData->_clazz = reinterpret_cast<TR_OpaqueClassBlock *>(J9_CLASS_FROM_METHOD(inlinedMethod));
+      reloPrivateData->_pointer = reinterpret_cast<uintptr_t>(inlinedMethod);
+      reloPrivateData->_needUnloadAssumption =
+            !reloRuntime->fej9()->sameClassLoaders(
+               reloPrivateData->_clazz,
+               reloRuntime->comp()->getCurrentMethod()->classOfMethod());
+
+      RELO_LOG(reloRuntime->reloLogger(), 6,"\tpreparePrivateData: pointer %p\n", reloPrivateData->_pointer);
+      }
+   else
+      {
+      reloPrivateData->_activatePointer = false;
+      reloPrivateData->_clazz = reinterpret_cast<TR_OpaqueClassBlock *>(-1);
+      reloPrivateData->_pointer = static_cast<uintptr_t>(-1);
+      reloPrivateData->_needUnloadAssumption = false;
+
+      RELO_LOG(reloRuntime->reloLogger(), 6,"\tpreparePrivateData: invalid site\n");
+      }
+   }
+
+int32_t
+TR_RelocationRecordInlinedMethodPointer::applyRelocation(TR_RelocationRuntime *reloRuntime, TR_RelocationTarget *reloTarget, uint8_t *reloLocation)
+   {
+   TR_RelocationRecordPointerPrivateData *reloPrivateData = &(privateData()->pointer);
+
+   /* Apply relocation */
+   reloTarget->storePointer((uint8_t *)reloPrivateData->_pointer, reloLocation);
+
+   if (reloPrivateData->_activatePointer)
+      {
+      /* Add class unload assumptions if needed */
+      if (reloPrivateData->_needUnloadAssumption)
+         reloTarget->addPICtoPatchPtrOnClassUnload(reloPrivateData->_clazz, reloLocation);
+
+      /* Add HCR assumptions if needed */
+      if (reloRuntime->options()->getOption(TR_EnableHCR))
+         {
+         createClassRedefinitionPicSite(
+                  reinterpret_cast<void *>(reloPrivateData->_pointer),
+                  reinterpret_cast<void *>(reloLocation),
+                  sizeof(uintptr_t),
+                  false,
+                  reloRuntime->comp()->getMetadataAssumptionList());
+
+         reloRuntime->comp()->setHasClassRedefinitionAssumptions();
+         }
+      }
```

Comment:
would you please add an assert on the else here that
    `reloPrivatData->_clazz == reinterpret_cast<TR_OpaqueClassBlock *>(-1)`

Reply:
Made changes, see force push https://github.com/eclipse/openj9/compare/d11a837eafbc2a6a2406558cce5d4defaaa09ac9..4a69430d1d1603f0a4a8666a6ca4ed21fedce751